File: 463/463.cpp
class Solution {
public:
    int islandPerimeter(vector<vector<int>>& grid) {
       int result = 0, row = grid.size(), col = grid[0].size();
       for(int i = 0; i < row; i++) {
        for(int j = 0; j < col; j++){
            if(grid[i][j]==1){
                result += 4;
                if(i>0 && grid[i-1][j]==1) result -=2;
                if(j>0 && grid[i][j-1] ==1) result -=2;
            }
        }
       } 
       return result;
    }
};

File: 463/463.java
class Solution {
  public int islandPerimeter(int[][] grid) {
      int result = 0;
      int row = grid.length;
      int col = grid[0].length;
      for(int i = 0; i<row; i++){
          for(int j = 0; j<col; j++){
              if(grid[i][j]==1){
                  if (j == 0 || grid[i][j - 1]==0) result++;
                  if (j == col - 1 || grid[i][j + 1]==0) result++;
                  if (i==0 || grid[i - 1][j]==0) result++;
                  if (i==row-1 || grid[i + 1][j]==0) result++;
              }
          }
      }
      return result;
  }
}